artemis-server/src/main/java/org/apache/activemq/artemis/core/paging/cursor/impl/PageCursorProviderImpl.java

```diff
@@ -74,6 +75,8 @@
 
    private final SoftValueLongObjectHashMap<PageCache> softCache;
 
+   private final LongObjectHashMap<PageCache> indexCacheMap;
```

Comment:
Wouldn't this break SoftReferences cleanup? meaning a page will be leaving forever.

I would use a SoftValueLongObjectHashMap here as well. In case memory is low, we want PageCache to be releasable. And this would make hard references on pageCache.

<<this is a question>>

Reply:
@clebertsuconic They have implemented a specialized `PageCache` that just hold the offset into the page file to read the data (kafka does something similar), not the full fate `PageCache` we are used to reference with Soft references, but I agree with Clebert that it has some GC costs (both due to the TreeMap and the Integer usage).
I think this could improved by using something primitive or (even better) by storing the offset in the page file in some paged message reference instead, wdyt?